### PR TITLE
added password validation

### PR DESCRIPTION
**Pull request by bhanuprakash-betsol**

### CODE REVIEW

File: middleware/validatePassword.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+schema
+.is().min(8)                        // Minimum length 8
+.is().max(20)                      // Maximum length 20
+.has().uppercase(1)               // Must have uppercase letters
+.has().lowercase(1)              // Must have lowercase letters
+.has().digits(1)                // Must have at least 1 digits
+.has().not().spaces()          // Should not have spaces
+.has().symbols(1)                 
+.is().not().oneOf(['Passw0rd', 'Password123']); // Blacklist these values
```
Comment (pramodjanardhana) [inline middleware/validatePassword.js:14]: blacklisted passwords can be taken from config.js

File: middleware/validatePassword.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+const passwords = function(req,res,next){
+    try{
+    const {newpassword} = req.body;
+    if(schema.validate(newpassword)){
+        next();
+    }
+        else{
+           let list = schema.validate(newpassword, { list: true });
+            return errorResponse(res,200,list)
```
Comment (pramodjanardhana) [inline middleware/validatePassword.js:25]: change the response code to 400 and what does the `list` contain? We don't want to give too much info on the error to the user

File: middleware/validatePassword.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        
+        }
+    }
+    catch(err){
+        console.log(err);
```
Comment (pramodjanardhana) [inline middleware/validatePassword.js:30]: send an error response with status code 500, message as `something went wrong while validating password`

and user logger.error instead on console.log

File: middleware/validatePassword.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+.is().max(20)                      // Maximum length 20
+.has().uppercase(1)               // Must have uppercase letters
+.has().lowercase(1)              // Must have lowercase letters
+.has().digits(1)                // Must have at least 1 digits
+.has().not().spaces()          // Should not have spaces
+.has().symbols(1)                 
+.is().not().oneOf(['Passw0rd', 'Password123']); // Blacklist these values
+
+
+const passwords = function(req,res,next){
```
Comment (pramodjanardhana) [inline middleware/validatePassword.js:17]: rename function to `validatePassword`

File: controllers/auth.js [hunk 5]
```diff
@@ -37,7 +37,7 @@ const adduser = async (req, res) => {
     });
 
     // only for development purpose this needs to removed later
-    //console.log(password);
+    console.log(password);
```
Comment (pramodjanardhana) [inline controllers/auth.js:40]: uncomment this log

File: middleware/validatePassword.js (hunk 2 above)
Comment (bhanuprakash-betsol) [reply middleware/validatePassword.js:25]: list will send the condition which are not matching

**Review by pramodjanardhana — APPROVED** [review]

**Pull request merged**